core/schemas/cachemetadata_test.go

```
@@ -0,0 +1,51 @@
+package schemas
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestCacheMetadataContextReturnsOwnedSnapshot verifies semantic cache metadata survives without aliases.
+func TestCacheMetadataContextReturnsOwnedSnapshot(t *testing.T) {
+	ctx := NewBifrostContext(nil, NoDeadline)
```

golangci-lint (2.13.2)

[error] SA1012: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use

(staticcheck)